### PR TITLE
[ALLUXIO-1831] Add Unit Tests for mount logic in FileSystemMaster

### PR DESCRIPTION
**Pull request by saltylin**
https://alluxio.atlassian.net/browse/ALLUXIO-1831

### CODE REVIEW

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java [hunk 1]
```diff
@@ -538,6 +542,112 @@ public void freeDirTest() throws Exception {
...
+    mFileSystemMaster.createDirectory(new AlluxioURI(alluxioURI, dirURI),
+        CreateDirectoryOptions.defaults().setPersisted(true));
+    mFileSystemMaster.unmount(alluxioURI);
+    AlluxioURI ufsDirURI = new AlluxioURI(ufsURI, dirURI);
+    File file = new File(ufsDirURI.toString());
+    Assert.assertTrue(file.exists());
+  }
+
+  /**
+   * Create a temporary ufs folder. The ufsPath must be a relative path since it's a temporary dir
```
Comment (jsimsa) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: Creates a temporary ...

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java [hunk 2]
```diff
@@ -538,6 +542,112 @@ public void freeDirTest() throws Exception {
...
+    AlluxioURI ufsURI = createTempUfsDir("ufs/hello");
+    mFileSystemMaster.mount(alluxioURI, ufsURI, MountOptions.defaults());
+    AlluxioURI shadowAlluxioURI = new AlluxioURI("/hello/shadow");
+    AlluxioURI anotherUfsURI = createTempUfsDir("ufs/hi");
+    mThrown.expect(InvalidPathException.class);
+    mFileSystemMaster.mount(shadowAlluxioURI, anotherUfsURI, MountOptions.defaults());
+  }
+
+  /**
+   * Tests mounting a prefix Ufs dir.
```
Comment (jsimsa) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: nit: use either ufs or UFS

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java [hunk 3]
```diff
@@ -538,6 +542,112 @@ public void freeDirTest() throws Exception {
...
+    AlluxioURI alluxioURI = new AlluxioURI("/hello");
+    mFileSystemMaster.mount(alluxioURI, ufsURI, MountOptions.defaults());
+    AlluxioURI preUfsURI = ufsURI.getParent();
+    AlluxioURI anotherAlluxioURI = new AlluxioURI("/hi");
+    mThrown.expect(InvalidPathException.class);
+    mFileSystemMaster.mount(anotherAlluxioURI, preUfsURI, MountOptions.defaults());
+  }
+
+  /**
+   * Tests mounting a suffix Ufs dir.
```
Comment (jsimsa) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: ditto

**Comment by jsimsa** [comment]
@saltylin thank you for adding the unit tests ... left some minor comments

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java (hunk 1 above)
Comment (saltylin) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: Updated!

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java (hunk 2 above)
Comment (saltylin) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: Updated!

File: core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java (hunk 3 above)
Comment (saltylin) [inline core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java]: Updated!

**Comment by saltylin** [comment]
@jsimsa Thank you very much for your review. I have addressed these comments.

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**